src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -133,14 +133,17 @@ public String toString() {
 
             // If there are two primary reads in the group pass them as a pair
             } else if (primaryReads.size()==2) {
-                final IndexPair<GATKRead> firstRead = primaryReads.get(0);
+                final GATKRead firstRead = primaryReads.get(0).getValue();
                 final IndexPair<GATKRead> secondRead = primaryReads.get(1);
-                final Pair pair = MarkDuplicatesSparkRecord.newPair(firstRead.getValue(), secondRead.getValue(), header, secondRead.getIndex(), scoringStrategy);
-                final Short readGroup = headerReadGroupIndexMap.getValue().get(firstRead.getValue().getReadGroup());
+                final Pair pair = MarkDuplicatesSparkRecord.newPair(firstRead, secondRead.getValue(), header, secondRead.getIndex(), scoringStrategy);
+                // Validate and add the read group to the pair
+                final Short readGroup = headerReadGroupIndexMap.getValue().get(firstRead.getReadGroup());
```

Comment:
could we add 2 simple tests that show that it explodes with the right errors on each of these cases